--- origsrc/libao-0.8.8/configure.ac	2007-05-24 05:51:05.000000000 -0500
+++ src/libao-0.8.8/configure.ac	2007-07-29 22:54:45.156250000 -0500
@@ -88,6 +88,11 @@
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore"
                 CFLAGS="-D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore"
                 PROFILE="-g -pg -D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore" ;;
+        *-cygwin*)
+                PLUGIN_LDFLAGS="-export-dynamic -module -avoid-version -no-undefined"
+                DEBUG="-g -Wall -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
+                CFLAGS="-O20 -ffast-math -D_REENTRANT -fsigned-char"
+                PROFILE="-pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char";;
         *)
                 PLUGIN_LDFLAGS="-export-dynamic -avoid-version"
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
@@ -142,6 +147,10 @@
 	DLOPEN_FLAG='(RTLD_LAZY)'
 	SHARED_LIB_EXT='.so'
 	;;
+    *cygwin*)
+	DLOPEN_FLAG='(RTLD_NOW | RTLD_GLOBAL)'
+	SHARED_LIB_EXT='.dll'
+	;;    
     *)
 	DLOPEN_FLAG='(RTLD_NOW | RTLD_GLOBAL)'
 	SHARED_LIB_EXT='.so'
--- origsrc/libao-0.8.8/src/plugins/nas/Makefile.am	2007-05-24 04:19:06.000000000 -0500
+++ src/libao-0.8.8/src/plugins/nas/Makefile.am	2007-07-29 22:47:26.312500000 -0500
@@ -5,7 +5,6 @@
 if HAVE_NAS
 
 nasltlibs = libnas.la
-nasldflags = -export-dynamic -avoid-version
 nassources = ao_nas.c
 
 else
@@ -21,7 +20,7 @@
 libdir = $(plugindir)
 lib_LTLIBRARIES = $(nasltlibs)
 
-libnas_la_LDFLAGS = $(nasldflags)
+libnas_la_LDFLAGS = @PLUGIN_LDFLAGS@
 libnas_la_LIBADD = @NAS_LIBS@
 libnas_la_SOURCES = $(nassources)
 
